src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/RecordingOptionsPatchHandler.java

```diff
@@ -0,0 +1,99 @@
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * #L%
+ */
+package com.redhat.rhjmc.containerjfr.net.web.handlers;
+
+import javax.inject.Inject;
+
+import com.redhat.rhjmc.containerjfr.core.RecordingOptionsCustomizer;
+import com.redhat.rhjmc.containerjfr.core.RecordingOptionsCustomizer.OptionKey;
+import com.redhat.rhjmc.containerjfr.net.AuthManager;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+class RecordingOptionsPatchHandler extends AbstractAuthenticatedRequestHandler {
+
+    static final String PATH = "/api/v1/targets/:targetId/recordingOptions";
+    private final RecordingOptionsCustomizer customizer;
+
+    @Inject
+    RecordingOptionsPatchHandler(AuthManager auth, RecordingOptionsCustomizer customizer) {
+        super(auth);
+        this.customizer = customizer;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.PATCH;
+    }
+
+    @Override
+    public String path() {
+        return PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    void handleAuthenticated(RoutingContext ctx) throws Exception {
+        MultiMap attrs = ctx.request().formAttributes();
+
+        if (attrs.contains("toDisk")) {
+            OptionKey.fromOptionName("toDisk")
+                    .ifPresent(key -> customizer.set(key, attrs.get("toDisk")));
+        }
+        if (attrs.contains("maxAge")) {
+            OptionKey.fromOptionName("maxAge")
+                    .ifPresent(key -> customizer.set(key, attrs.get("maxAge")));
+        }
+        if (attrs.contains("maxSize")) {
+            OptionKey.fromOptionName("maxSize")
+                    .ifPresent(key -> customizer.set(key, attrs.get("maxSize")));
+        }
+        ctx.response().setStatusCode(200);
```

Comment:
I wonder if it would make sense to respond with 200 and then send back the updated map, or respond `204` and no response body? If the client doesn't care about the updated value then they can simply ignore it.

Reply:
Yes, returning the updated map is good, do I need to convert it to json when sending it back?

Reply:
Yes.

Reply:
Should work now!